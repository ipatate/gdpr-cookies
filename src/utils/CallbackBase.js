// @flow
export const createScript = (src: string): void => {
  const tag: HTMLScriptElement = document.createElement('script');
  tag.type = 'text/javascript';
  tag.async = true;
  tag.src = src;
  const s: HTMLScriptElement = document.getElementsByTagName('script')[0];
  const parent: ?Node = s.parentNode;
  if (parent) {
    parent.insertBefore(tag, s);
  }
};
