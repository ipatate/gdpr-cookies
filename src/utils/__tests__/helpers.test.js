import {createScript, createStyle, createIframe} from '../CallbackBase';

test('test Script', () => {
  const script = createScript('myscript.js');
  expect(script).toMatchSnapshot();
});

test('test Style', () => {
  const style = createStyle('mystyle.css');
  expect(style).toMatchSnapshot();
});

test('test iframe', () => {
  const iframe = createIframe('id', {href: 'lol', width: '200px'});
  expect(iframe).toMatchSnapshot();
});
