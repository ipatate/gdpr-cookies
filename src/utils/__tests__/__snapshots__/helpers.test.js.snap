// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Script 1`] = `
<script
  src="myscript.js"
  type="text/javascript"
/>
`;

exports[`test Style 1`] = `
<link
  href="mystyle.css"
  rel="stylesheet"
  type="text/css"
/>
`;

exports[`test iframe 1`] = `
<iframe
  href="lol"
  width="200px"
/>
`;
